### PR TITLE
add support for the Sign and Verify restful API

### PR DESCRIPTION
**Pull request by Liang1XZhao**
this restful API will exposed to public as:
- POST <ehsm_srv_address>/ehsm?Action=Sign
- POST <ehsm_srv_address>/ehsm?Action=Verify
add the corresponding test cases.
test: passed the unittest

Signed-off-by: Liang1XZhao <liang1x.zhao@intel.com>

### CODE REVIEW

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 1]
```diff
@@ -2,157 +2,156 @@
...
+
+    params["appid"] = appid
+    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
+    params["timestamp"] = int(time.time())
+
+    query_string = urllib.parse.urlencode(params)
+    query_string += '&app_key=' + appkey
+    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), query_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8')
+
+    params["sign"] = sign
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:27]: signature should contain all params including the payload.

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 2]
```diff
@@ -2,157 +2,156 @@
...
+import urllib.parse
+
+appid = '202112101919'
+appkey = '202112345678'
+
+def test_params(payload):
+    params = OrderedDict()
+
+    params["appid"] = appid
+    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:20]: is "str" required?

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 3]
```diff
@@ -2,157 +2,156 @@
...
+import time
+import random
+import hmac
+from hashlib import sha256
+import os
+from collections import OrderedDict
+import urllib.parse
+
+appid = '202112101919'
+appkey = '202112345678'
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:14]: hardcode a 32B key for HMAC-SHA256

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 4]
```diff
@@ -2,157 +2,156 @@
...
+
+def test_params(payload):
+    params = OrderedDict()
+
+    params["appid"] = appid
+    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
+    params["timestamp"] = int(time.time())
+
+    query_string = urllib.parse.urlencode(params)
+    query_string += '&app_key=' + appkey
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:24]: Do not transfer app_key, it's used by HMAC-SHA256 calculation only.

File: ehsm_kms_service/ehsm_kms_server.js [hunk 5]
```diff
@@ -43,7 +49,24 @@ const _checkSign = function (req,res,next){
...
+  let str = '';
+  let sign_parmas = {appid, nonce, timestamp} 
+  for(var k in sign_parmas){
+    if(!str) {
+      str += k + '=' + sign_parmas[k]
+    } else{
+      str += '&' + k + '=' + sign_parmas[k]
+    }
+  }
+  str += '&app_key=' + test_app_key;
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_server.js:63]: app_key is for internal hmac-sha256 calculation only. 
you can hardcode a 32B app key in .js and .py, but not transfer btw them.

File: ehsm_kms_service/ehsm_kms_server.js [hunk 6]
```diff
@@ -43,7 +49,24 @@ const _checkSign = function (req,res,next){
...
+  let local_sign = crypto.createHmac('sha256',test_app_key).update(str, 'utf8').digest('base64');
+  if(sign != local_sign) {
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_server.js:65]: BTW, are sign and local_sign in lower case or upper case?

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 3 above)
Comment (yang8621) [reply ehsm_kms_service/test/test_kms_with_rest.py:14]: ignore this comment.

File: ehsm_kms_service/ehsm_kms_server.js (hunk 6 above)
Comment (syan10) [reply ehsm_kms_service/ehsm_kms_server.js:65]: for the parameter sign, suggest to do it in next commit.

**Review by syan10 — APPROVED** [review]
you can add a task to refine the parameter sign in next commit, this pr could be merged in this time.

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 1 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_kms_with_rest.py:27]: It will be fix.

**Pull request merged**